libraries/botbuilder-lg/src/expander.ts

```diff
@@ -607,8 +621,14 @@ export class Expander extends AbstractParseTreeVisitor<string[]> implements LGTe
         return undefined;
     };
 
+    private readonly evaluateWithTemplates = (templateName: string, templates: Templates): any => (args: readonly any[]): any => {
+        const newScope = this.constructScope(templateName, args.slice(0), templates.allTemplates);
+
+        return templates.evaluate(templateName, newScope);
+    };
+
     private readonly templateEvaluator = (templateName: string): any => (args: readonly any[]): any => {
-        const newScope: any = this.constructScope(templateName, Array.from(args));
+        const newScope: any = this.constructScope(templateName, Array.from(args), this.templates.allTemplates);
```

Comment:
Why do we need `Array.from()`?

Reply:
It is a shallow copy of the original read only array to avoid the in place change. 
Same effect with: `args.slice(0)`, or `[...args]`, or `args.concat()`